File: frontend/src/types/strategy-template.ts
import { Block, Connection } from "./canvas";

export interface StrategyTemplate {
  id: string;
  name: string;
  description: string;
  logic_summary: string;
  use_cases: string[];
  parameter_ranges: Record<string, string>;
  asset: string;
  timeframe: string;
  difficulty: string;
  sort_order: number;
  teaches_description: string | null;
  created_at: string;
}

export interface StrategyTemplateDetail extends StrategyTemplate {
  definition_json: {
    blocks: Block[];
    connections: Connection[];
    meta: Record<string, unknown>;
  };
}

File: backend/app/schemas/strategy_template.py
from datetime import datetime
from typing import Optional
from uuid import UUID
from pydantic import BaseModel


class TemplateResponse(BaseModel):
    id: UUID
    name: str
    description: str
    logic_summary: str
    use_cases: list[str]
    parameter_ranges: dict[str, str]
    asset: str
    timeframe: str
    difficulty: str
    sort_order: int
    teaches_description: Optional[str] = None
    created_at: datetime


class TemplateDetailResponse(TemplateResponse):
    definition_json: dict
